bindings/pyroot/cppyy/CPyCppyy/src/Executors.cxx

```diff
@@ -604,8 +605,12 @@ PyObject* CPyCppyy::InstanceExecutor::Execute(
     if (!pyobj)
         return nullptr;
 
-// python ref counting will now control this object's life span
-    ((CPPInstance*)pyobj)->PythonOwns();
+// python ref counting will now control this object's life span (if it is
+// an exception, no need to call PythonOwns on the internal CPPInstance,
+// since it is flagged as kIsValue)
+    if (!CPPExcInstance_Check(pyobj))
```

Comment:
But does it hurt to call `((CPPInstance*)pyobj)->PythonOwns();` if it's an `CPPExcInstance`? That's not clear to me here.

Reply:
Yes, because they are two different classes (non inheritance-related) and thus calling `PythonOwns` on a `CPPExcInstance` causes it's corruption.

Reply:
I think this is not really clear from the comment in the code. Probably it's worth to clarify?

Reply:
Actually this check has been removed in the new version of the commit, in its place there is a comment that explains how the object is going to be deleted.